### PR TITLE
Move to v0.7.0

### PR DESCRIPTION
**Pull request by sullof**
- [x] Removing Canonical in favor of hardcoding the registries' addresses.
- [x] Move from nameId to basic implementation address to identify a plugin

### CODE REVIEW

File: contracts/manager/CrunaManager.sol [hunk 1]
```diff
@@ -22,86 +22,129 @@ import {Deployed} from "../utils/Deployed.sol";
...
     return _allPlugins[index];
   }
 
-  /// @dev see {ICrunaManager-migrate}
+  /**
+   * @dev During an upgrade allows the manager to perform adjustments if necessary.
+   * The parameter is the version of the manager being replaced. This will allow the
+   * new manager to know what to do to adjust the state of the new manager.
+   */
   function migrate(uint256 /* version */) external virtual override {
```
Comment (andyscraven) [inline contracts/manager/CrunaManager.sol:78]: migrate is a fairly generic word, maybe pluginMigrate

File: contracts/token/CrunaProtectedNFT.sol [hunk 2]
```diff
@@ -93,7 +111,20 @@ abstract contract CrunaProtectedNFT is ICrunaProtectedNFT, IVersioned, IERC6454,
...
+   * @notice Initialize the NFT
+   * @param managerAddress_ The address of the manager
+   * @param progressiveTokenIds_ If true, the tokenIds will be progressive
+   * @param nextTokenId_ The next tokenId to be used.
+   * If progressiveTokenIds_ == true and the project must reserve some tokens to
+   * special addresses, community, etc. You set the nextTokenId_ to the first not reserved token.
+   * Be careful, your function minting by tokenId MUST check that the tokenId is
+   * not higher than nextTokenId. If not, when trying to mint tokens by amount, as soon as
+   * nextTokenId reaches the minted tokenId, the function will revert, blocking any future minting.
+   * If you code may risk so, set a function that allow you to correct the nextTokenId to skip
```
Comment (andyscraven) [inline contracts/token/CrunaProtectedNFT.sol]: "If you code may risk so" does not make sense.

File: contracts/token/CrunaProtectedNFTTimeControlled.sol [hunk 3]
```diff
@@ -48,7 +48,13 @@ abstract contract CrunaProtectedNFTTimeControlled is CrunaProtectedNFT, Timelock
     return super.supportsInterface(interfaceId);
   }
 
-  /// @dev see {CrunaProtectedNFT-_canManage}
+  /**
+   * @notice Specify if the caller can call some function.
+   * Must be overridden to specify who can manage changes during initialization and later
+   * @param isInitializing If true, the function is being called during initialization, if false,
+   * it is supposed to the called later. A time controlled NFT can allow the admin to call some
```
Comment (andyscraven) [inline contracts/token/CrunaProtectedNFTTimeControlled.sol]: it is supposed to BE called later

File: contracts/utils/SignatureValidator.sol [hunk 4]
```diff
@@ -59,22 +59,46 @@ abstract contract SignatureValidator is ISignatureValidator, EIP712, Context {
...
+   * @param hash The hash of the signature.
+   */
   function isSignatureUsed(bytes32 hash) external view override returns (bool) {
     return _usedSignatures[hash] == 1;
   }
 
-  /// @dev see {ISignatureValidator-recoverSigner}
+  /**
+   * @dev This function validates a signature trying to be as flexible as possible.
+   * As long as called inside the same contract, the cost adding some more parameters is negligible.
```
Comment (andyscraven) [inline contracts/utils/SignatureValidator.sol]: the cost OF adding some more parameter

File: contracts/manager/CrunaManager.sol (hunk 1 above)
Comment (sullof) [reply contracts/manager/CrunaManager.sol:78]: Yes, but that is in the manager, so we do not risk any conflicts. The function in ICrunaService and ICrunaManagedService are the ones that must avoid conflicts.

File: contracts/token/CrunaProtectedNFT.sol (hunk 2 above)
Comment (sullof) [reply contracts/token/CrunaProtectedNFT.sol]: It is a typo, that is _If your code may..._. I will fix it

**Review by andyscraven — APPROVED** [review]

**Pull request merged**